transformations/insert_abbreviation/transformation.py

```diff
@@ -0,0 +1,80 @@
+import itertools
+import random
+import sys
+from interfaces.SentenceOperation import SentenceOperation
+from tasks.TaskTypes import TaskType
+
+sys.path.append("./transformations/insert_abbreviation")
+import grammaire
+
+
+def readfile(file):
+    with open(file, encoding='utf8') as input:
+        lines=input.readlines()
+    return lines
+
+def load_rules(file):
+    with open(file, encoding='utf8') as input:
+        str_rules = input.read()
+    return str_rules
+
+"""
+Abbreviations
+"""
+
+
+class AbbreviationInsertion(SentenceOperation):
+    tasks = [
+        TaskType.TEXT_CLASSIFICATION,
+        TaskType.TEXT_TO_TEXT_GENERATION,
+        TaskType.TEXT_TAGGING,
+    ]
+    languages = ["en", "fr"]
+
+    def __init__(self, seed=0, max_outputs=1):
+        rulefile_en = "./transformations/insert_abbreviation/replacement_rules_en.txt"
+        rulefile_fr = "./transformations/insert_abbreviation/replacement_rules_fr.txt"
+        super().__init__(seed, max_outputs=max_outputs)
+        rules_en = load_rules(rulefile_en)
+        rules_fr = load_rules(rulefile_fr)
+        # First we compile our rules...
+        self.grammar_en = grammaire.compile(rules_en)
+        self.grammar_fr = grammaire.compile(rules_fr)
+
+    def generate(self, sentence: str):
+        results = grammaire.parse(sentence, self.grammar_en)
```

Comment:
It might be better to separate the ones for French and the English tranformation. Also, in this implementation doesn't look like you are using the French one. You don't need to create a separate folder - just within this transformation.py, you could have `AbbreviationInsertionEn`  & `AbbreviationInsertionFr`